File: ios/NestMind/Sources/Core/Supabase/SupabaseService.swift
import Foundation
import Supabase

final class SupabaseService {
  let client: SupabaseClient

  init(config: AppConfig) {
    client = SupabaseClient(
      supabaseURL: config.supabaseURL,
      supabaseKey: config.supabasePublishableKey
    )
  }
}

File: apps/android/app/src/main/java/com/nestmind/app/core/auth/SupabaseService.kt
package com.nestmind.app.core.auth

import com.nestmind.app.core.config.AppConfig
import io.github.jan.supabase.SupabaseClient
import io.github.jan.supabase.auth.Auth
import io.github.jan.supabase.createSupabaseClient
import io.github.jan.supabase.storage.Storage
import java.io.Closeable

class SupabaseService(config: AppConfig) : Closeable {
    val client: SupabaseClient = createSupabaseClient(
        supabaseUrl = config.supabaseUrl,
        supabaseKey = config.supabaseAnonKey
    ) {
        install(Auth) {
            scheme = "nestmind"
            host = "auth-callback"
        }
        install(Storage)
    }

    override fun close() {
        client.close()
    }
}
